### PR TITLE
Port : Add Notification For `BOM_VALIDATION_FAILED`

### PR DESCRIPTION
**Pull request by sahibamittal**
### Description

If uploaded BOM is invalid, dispatches a notification with InvalidBomProblemDetails before throwing the respective exception

### Addressed Issue

Backport change : https://github.com/DependencyTrack/hyades/issues/1358

### Checklist

- [x] I have read and understand the [contributing guidelines]
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have updated the [migration changelog] accordingly
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation] accordingly

[contributing guidelines]: ../CONTRIBUTING.md#pull-requests
[documentation]: https://dependencytrack.github.io/hyades/latest/development/documentation/
[migration changelog]: https://dependencytrack.github.io/hyades/latest/development/database-migrations/

### CODE REVIEW

File: src/main/proto/org/dependencytrack/notification/v1/notification.proto [hunk 1]
```diff
@@ -80,6 +81,13 @@ message BomProcessingFailedSubject {
   string token = 4;
 }
 
+message BomValidationFailedSubject {
+  Project project = 1;
+  Bom bom = 2;
+  repeated string errors = 3;
+  string format = 4;
```
Comment (nscuro) [inline src/main/proto/org/dependencytrack/notification/v1/notification.proto:88]: Do we need this? The `Bom` message already has a `format` field.

File: src/test/java/org/dependencytrack/parser/dependencytrack/NotificationModelConverterTest.java [hunk 2]
```diff
@@ -428,6 +431,35 @@ public void testConvertBomProcessingFailedNotification() throws Exception {
         assertThat(subject.getCause()).isEqualTo("just because");
     }
 
+    @Test
+    public void testConvertBomValidationFailedNotification() throws Exception {
+        final org.dependencytrack.model.Project project = createProject();
+        final var token = UUID.randomUUID();
```
Comment (nscuro) [inline src/test/java/org/dependencytrack/parser/dependencytrack/NotificationModelConverterTest.java:437]: `token` appears to be unused (see Codacy warning). Can we remove this?

File: src/test/java/org/dependencytrack/resources/v1/BomResourceTest.java [hunk 3]
```diff
@@ -1226,6 +1244,15 @@ public void uploadBomInvalidXmlTest() {
                   ]
                 }
                 """);
+
+        assertConditionWithTimeout(() -> kafkaMockProducer.history().size() == 1, Duration.ofSeconds(5));
```
Comment (nscuro) [inline src/test/java/org/dependencytrack/resources/v1/BomResourceTest.java:1248]: You can replace `assertConditionWithTimeout` with a simple `assertThat`. Events will already have been sent at the point of the assert, so no waiting necessary.

```suggestion
        assertThat(kafkaMockProducer.history()).hasSize(1);
```

**Review by nscuro — CHANGES_REQUESTED** [review]

File: src/main/proto/org/dependencytrack/notification/v1/notification.proto (hunk 1 above)
Comment (sahibamittal) [reply src/main/proto/org/dependencytrack/notification/v1/notification.proto:88]: yes mom already has format info, it can be removed 👍🏼

**Review by nscuro — APPROVED** [review]

**Pull request merged**